build-tools/packages/build-tools/src/type-test-generator/typeTestUtils.ts

```diff
@@ -0,0 +1,289 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { readJsonSync } from "fs-extra";
+import path from "node:path";
+import { IExtractorConfigPrepareOptions } from "@microsoft/api-extractor";
+import { existsSync } from "node:fs";
+import { Project, SourceFile } from "ts-morph";
+import { BrokenCompatTypes } from "../common/fluidRepo";
+import { buildTestCase, TestCaseTypeData } from "../typeValidator/testGeneration";
+import { getFullTypeName, getNodeTypeData, TypeData } from "../typeValidator/typeData";
+import { PackageJson } from "../common/npmPackage";
+
+/**
+ * Checks the package object to verify that the specified dependency exists
+ * @param packageObject - package.json object
+ * @param dependencyName - the dependency to check for in the package object
+ * @remarks Information about the previous package from the package.json is not needed,
+ * but error if it's missing since it's nice to separate errors for the dep missing here vs not installed.
+ * This ensures that a critical dependency (the previous package version) is correctly declared in the project's package.json.
+ */
+export function ensureDevDependencyExists(
+	packageObject: PackageJson,
+	dependencyName: string,
+): void {
+	const dependencyVersion = packageObject?.devDependencies?.[dependencyName];
+	if (typeof dependencyVersion !== "string") {
+		throw new Error(`Did not find devDependency ${dependencyName} in package.json`);
+	}
+}
+
+/**
+ * Fetches the path of the previous package.json or throws an error if not found.
+ * @param previousBasePath - A string representing the path to the root of a package
+ */
+export function getPreviousPackageJsonPath(previousBasePath: string): string {
+	const previousPackageJsonPath = path.join(previousBasePath, "package.json");
+	if (!existsSync(previousPackageJsonPath)) {
+		throw new Error(`${previousPackageJsonPath} not found.`);
+	}
+	return previousPackageJsonPath;
+}
+
+/**
+ * Attempts to retrieve a specified type of rollup file path for type definitions from the API Extractor config.
+ * @param rollupType - The type of rollup file path to retrieve (ex: "alpha", "beta", "public").
+ * @param extractorConfig - The API Extractor config object.
+ * @returns The path to the type definitions file for the specified rollupType, or undefined if it cannot be found.
+ * @throws If api-extractor config cannot be loaded.
+ */
+export function getTypeRollupPathFromExtractorConfig(
```

Comment:
Nit: something that might enhance easy legibility here would be to call this `tryGet...` - I usually like that naming when a function doesn't throw in its failure mode. That helps imply to users that they have to do something special on their end to handle the non-result (in this case undefined).